pkg/query/querier.go

```diff
@@ -114,12 +114,45 @@ func (q *querier) selectSingle(conn *grpc.ClientConn, ms ...storepb.LabelMatcher
 	return &storeSeriesSet{series: resp.Series, i: -1, mint: q.mint, maxt: q.maxt}, nil
 }
 
-func (*querier) LabelValues(name string) ([]string, error) {
-	return nil, errors.New("not implemented")
+func (q *querier) LabelValues(name string) ([]string, error) {
+	var (
+		mtx sync.Mutex
+		all []string
+		// TODO(bplotka): errgroup will fail the whole query on the first encountered error.
+		// Add support for partial results/errors.
+		g errgroup.Group
+	)
+
+	for _, s := range q.stores {
+		g.Go(func() error {
+			values, err := q.labelValuesSingle(s.Conn(), name)
+			if err != nil {
+				return err
+			}
+
+			mtx.Lock()
+			all = append(all, values...)
+			mtx.Unlock()
+
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
+		return []string(nil), err
+	}
+	return all, nil
```

Comment:
This will return a lot of duplicates.

We should check in `labelValuesSingle` that they are sorted and sort them if not already. Then here we should merge all sets together with like [here](https://github.com/prometheus/tsdb/blob/master/querier.go#L62-L77). Probably best to copy that code to `fanin.go` as well and eventually care about making the functionality exported in `prometheus/tsdb`.